tlslite/keyexchange.py

```diff
@@ -294,10 +298,30 @@ def processServerKeyExchange(self, srvPublicKey, serverKeyExchange):
 
     def makeClientKeyExchange(self):
         """Create client key share for the key exchange"""
-        cke = super(DHE_RSAKeyExchange, self).makeClientKeyExchange()
+        cke = super(ADHKeyExchange, self).makeClientKeyExchange()
         cke.createDH(self.dh_Yc)
         return cke
 
+# the DHE_RSA part comes from IETF ciphersuite names, we want to keep it
+#pylint: disable = invalid-name
+class DHE_RSAKeyExchange(ADHKeyExchange):
+    """
+    Handling of ephemeral Diffe-Hellman Key exchange
+
+    NOT stable API, do NOT use
+    """
+
+    def __init__(self, cipherSuite, clientHello, serverHello, privateKey):
+        super(DHE_RSAKeyExchange, self).__init__(cipherSuite, clientHello,
+                                                 serverHello, privateKey)
+#pylint: enable = invalid-name
+
+    def makeServerKeyExchange(self, sigHash=None):
+        """Prepare server side of key exchange with selected parameters"""
+        super(DHE_RSAKeyExchange, self).makeServerKeyExchange()
+        self.signServerKeyExchange(self.serverKeyExchange, sigHash)
```

Comment:
because adding attributes to objects outside constructors is bad form (you can see landscape test complaining about it)

and `makeServerKeyExchange()` needs to return (and already returns) a ServerKeyExchange object, so there is no need to pass this SKE through a field, it just makes code more complex